src/storage/sql/mod.rs

```diff
@@ -166,149 +181,166 @@ impl SqlStorage {
                     .and(tags::value.ne_all(&snippet.tags)),
             ),
         )
-        .execute(conn)?;
+        .execute(conn)
+        .await?;
 
         Ok(())
     }
 }
 
+#[rocket::async_trait]
 impl Storage for SqlStorage {
-    fn create(&self, snippet: &Snippet) -> Result<Snippet, StorageError> {
-        let mut conn = self.pool.get()?;
+    async fn create(&self, snippet: &Snippet) -> Result<Snippet, StorageError> {
+        let mut conn = self.pool.get().await?;
         conn.transaction::<_, StorageError, _>(|conn| {
-            // insert the new snippet row first to get the generated primary key
-            let snippet_id = self.insert_snippet(conn, snippet)?;
-            // insert the associated changesets
-            self.upsert_changesets(conn, snippet_id, snippet)?;
-            // insert the associated tags
-            self.upsert_tags(conn, snippet_id, snippet)?;
+            async move {
```

Comment:
What's the purpose of `async move`?

Reply:
`async` blocks become futures. `move` works similarly to how it works in closures: values that are referenced from inside closures or `async` blocks can either be borrowed (immutably or mutably) or moved; what exactly happens depends on how we use those values (e.g. whether we call methods on them that require `self` or `&self`); `move` tells the compiler that the values must be moved and not borrowed.

All the said, that's not needed here:

* `conn` is a _reference_ passed to the closure
* `snippet` is also reference

So we don't really need `move`. I think I just copied it from the Diesel docs :P Let me clean this up!